src/parsing.jl

```diff
@@ -66,15 +66,25 @@ variable defined elsewhere.
 """
 function parse_expr_to_symbolic end
 
-parse_expr_to_symbolic(x::Number, mod::Module) = x
+parse_expr_to_symbolic(x::Number, mod::Union{Module, AbstractDict}) = x
+
 function parse_expr_to_symbolic(x::Symbol, mod::Module)
   if isdefined(mod, x)
     getfield(mod, x)
   else
     (@variables $x)[1]
   end
 end
-function parse_expr_to_symbolic(ex, mod::Module)
+
+function parse_expr_to_symbolic(x::Symbol, dict::AbstractDict)
+  if haskey(dict, x)
+    dict[x]
+  else
+    (@variables $x)[1]
+  end
+end
+
+function parse_expr_to_symbolic(ex, mod::Union{Module,AbstractDict})
     if ex.head == :call
         if isdefined(mod, ex.args[1])
             return getfield(mod,ex.args[1])(parse_expr_to_symbolic.(ex.args[2:end],(mod,))...)
```

Comment:
I'm not sure why it doesn't error - we probably don't hit this case - but this method doesn't handle `mod::AbstractDict`?